### PR TITLE
Updated to Neo 3.6.0

### PR DESCRIPTION
**Pull request by cschuchardt88**
Porting to Neo 3.6.0

brought up to coding standards well.

Ported and updated `neo-blockchaintoolkit-library` to this repo. Since development has stopped for https://github.com/N3developertoolkit/neo-blockchaintoolkit-library and developer left.

Changed all configurations since repo is outdated and some projects are dead.

### CODE REVIEW

**Comment by cschuchardt88** [comment]
@dotnet-policy-service agree

**Comment by cschuchardt88** [comment]
Only thing is I don't know if your Github actions will build it right, I'll have a look at that.

**Comment by cschuchardt88** [comment]
@gsmachado @steven1227  does anyone monitor this repo anymore? I would like to knock this out and some issues.

**Comment by steven1227** [comment]
> @gsmachado @steven1227 does anyone monitor this repo anymore? I would like to knock this out and some issues.

Yes. We are looking into it as well.

**Comment by ixje** [comment]
I think it's hard to follow what actually changed when also reformatting almost every file. Those should really be separate PRs. Also, why there Plugin binaries checked in?

**Comment by cschuchardt88** [comment]
Look at

https://github.com/neo-project/neo-express/blob/e75c812d238f8727b844a82d0679e22d35df7319/src/neoxp/neoxp.csproj#L15-L28

Plugins have to come from somewhere. The configurations make assumptions that you have these files, along with other things, Like directories outside this repo's directory.

Like:

https://github.com/neo-project/neo-express/blob/e75c812d238f8727b844a82d0679e22d35df7319/src/Directory.Build.props#L21-L27

```xml
<add key="neo-monorepo" value="https://neomonorepopackages.blob.core.windows.net/packages/index.json" />
```
image

Here are the thinks i changed.
- Naming conventions
- Moved neo-blockchaintoolkit-library to this repo (no more developer and dead repo)
  - Fixed code and updated Neo version 3.6.0
- Removed unnecessary or broken configurations 
- Added .editorconfig
- Updated `nuget.config`
- Update Github actions to dotnet 7
- Unused or unnecessary packages from projects.

It's time to stop messing around and get this repo going again. I have tons of time on my hands. I tried many times to get in touch with core. I had someone on inside say developer left and it's not maintained anymore and this would be a place to start. So I'm just getting this repo up for public access... so anyone can start working on it. Should of been that way in the 1st place. That is one of the reasons why this repo was always dead. No support!

**Comment by Ashuaidehao** [comment]
Good work! We are planning reactive https://github.com/N3developertoolkit/neo-blockchaintoolkit-library, so moving bctk repo into neo-express is not most urgent task right now. Maybe we can continue this process after neo-express v3.6 release. If you have any improvement sugguestions, welcome issues and PRs in our repos.

**Comment by cschuchardt88** [comment]
There was some interface changes in `neo-blockchaintoolkit-library` on that respo. So i went ahead and implemented the missing methods on this repo for `neo-blockchaintoolkit-library`. Also i changed the naming conventions of the methods to meet dotnet code standards (thats why i added `.editorconfig`). If you would like i can put in a PR to get `neo-blockchaintoolkit-library` repo working with `neo-express` repo. Plus i use Visual Studio, not Visual Studio Code.

**Comment by ixje** [comment]
> I had someone on inside say developer left and it's not maintained anymore and this would be a place to start. So I'm just getting this repo up for public access... so anyone can start working on it. Should of been that way in the 1st place. That is one of the reasons why this repo was always dead.

I think it's public knowledge by now that the author left the NEO community for a different adventure. We're waiting for @JohndeVadoss to come with the new plan who's going to maintain it. There's more than enough things to fix and improve in this project, but without active accountable person we're just stuck.

**Comment by steven1227** [comment]
> > I had someone on inside say developer left and it's not maintained anymore and this would be a place to start. So I'm just getting this repo up for public access... so anyone can start working on it. Should of been that way in the 1st place. That is one of the reasons why this repo was always dead.
> 
> I think it's public knowledge by now that the author left the NEO community for a different adventure. We're waiting for @JohndeVadoss to come with the new plan who's going to maintain it. There's more than enough things to fix and improve in this project, but without an active accountable person we're just stuck.

I discussed with NGDseattle and there will be a person working on it in near future. For now, our highest priority is to update and release it to compatible with 3.6

**Comment by cschuchardt88** [comment]
>For now, our highest priority is to update and release it to compatible with 3.6

Waiting on https://github.com/N3developertoolkit/neo-blockchaintoolkit-library/pull/91 to merge

**Comment by cschuchardt88** [comment]
@steven1227 `neo-express` relies on `neo-blockchaintoolkit-library`. I had to update that too. That why i added `neo-blockchaintoolkit-library`  to this repo instead. You can use this PR if you like to get it going until someone merges at that repo. Or at least test it out.

**Comment by Liaojinghui** [comment]
my bad for changing the status, cont control myself to click it when there is a button.
